lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -775,7 +775,10 @@ static SmallVector<SubfieldOp> getAllFieldAccesses(Value structValue,
   for (auto op : structValue.getUsers()) {
     assert(isa<SubfieldOp>(op));
     auto fieldAccess = cast<SubfieldOp>(op);
-    if (fieldAccess.fieldname() == field) {
+    auto elemIndex =
+        fieldAccess.input().getType().cast<BundleType>().getElementIndex(field);
+    if (elemIndex.hasValue() &&
+        fieldAccess.fieldIndex() == elemIndex.getValue()) {
```

Comment:
Can't this `assert(elemIndex.hasValue())`?   The case where this fails is invalid IR right?

Reply:
What if the `field` argument to this `getAllFieldAccesses` is not even an element of the bundle ? Seems like need not be invalid IR, if `getAllFieldAccesses` is called with incorrect string field name.